File: utils/trie/triemap/trieset.go
package triemap

type TrieSet struct {
	trieMap *TrieMap[int]
}

func (ts *TrieSet) Put(key string) {
	(*ts.trieMap).Put(key, 0)
}
func (ts *TrieSet) Remove(key string) {
	(*ts.trieMap).Remove(key)
}
func (ts *TrieSet) ContainsKey(key string) bool {
	return (*ts.trieMap).ContainsKey(key)
}
func (ts *TrieSet) ShortestPrefixOf(query string) string {
	return (*ts.trieMap).ShortestPrefixOf(query)
}
func (ts *TrieSet) LongestPrefixOf(query string) string {
	return (*ts.trieMap).LongestPrefixOf(query)
}
func (ts *TrieSet) KeysWithPrefix(prefix string) []string {
	return (*ts.trieMap).KeysWithPrefix(prefix)
}
func (ts *TrieSet) HasKeyWithPrefix(prefix string) bool {
	return (*ts.trieMap).HasKeyWithPrefix(prefix)
}
func (ts *TrieSet) KeysWithPattern(pattern string) []string {
	return (*ts.trieMap).KeysWithPattern(pattern)
}
func (ts *TrieSet) HasKeyWithPattern(pattern string) bool {
	return (*ts.trieMap).HasKeyWithPattern(pattern)
}
func (ts *TrieSet) Len() int {
	return (*ts.trieMap).Len()
}

func GetTrieSet() *TrieSet {
	x := Default[int]()
	var t TrieMap[int] = &x

	return &TrieSet{
		trieMap: &t,
	}
}

File: 1804. Implement Trie II (Prefix Tree)/src/TrieSet.java
import java.util.List;
import java.util.function.Function;

/**
 * Your Trie object will be instantiated and called as such:
 * Trie obj = new Trie();
 * obj.insert(word);
 * boolean param_2 = obj.search(word);
 * boolean param_3 = obj.startsWith(prefix);
 */

class TrieSet {
    private final TrieMap<Object> map;
    public TrieSet(){
        map = new CharTrieMap<>();
    }
    public TrieSet(int size, Function<Character, Integer> charToIntConvertor, Function<Integer, Character> intToCharConvertor){
        map = new CharTrieMap<Object>(size, charToIntConvertor, intToCharConvertor);
    }
    
    public void add(String key) {
        map.put(key, new Object());
    }
    
    public void remove(String key) {
        map.remove(key);
    }
    
    public boolean contains(String key) {
        return map.containsKey(key);
    }
    
    public String shortestPrefixOf(String query) {
        return map.shortestPrefixOf(query);
    }
    
    public String longestPrefixOf(String query) {
        return map.longestPrefixOf(query);
    }
    
    public List<String> keysWithPrefix(String prefix) {
        return map.keysWithPrefix(prefix);
    }
    
    public boolean hasKeyWithPrefix(String prefix) {
        return map.hasKeyWithPrefix(prefix);
    }
    
    public List<String> keysWithPattern(String pattern) {
        return map.keysWithPattern(pattern);
    }
    
    public boolean hasKeyWithPattern(String pattern) {
        return map.hasKeyWithPattern(pattern);
    }
    
    public int size() {
        return map.size();
    }
}

